server/src/main/java/org/elasticsearch/search/aggregations/pipeline/bucketselector/BucketSelectorPipelineAggregator.java

```diff
@@ -82,34 +83,41 @@ public InternalAggregation reduce(InternalAggregation aggregation, ReduceContext
                 (InternalMultiBucketAggregation<InternalMultiBucketAggregation, InternalMultiBucketAggregation.InternalBucket>) aggregation;
         List<? extends InternalMultiBucketAggregation.InternalBucket> buckets = originalAgg.getBuckets();
 
-        ExecutableScript.Factory factory = reduceContext.scriptService().compile(script, ExecutableScript.AGGS_CONTEXT);
         List<InternalMultiBucketAggregation.InternalBucket> newBuckets = new ArrayList<>();
-        for (InternalMultiBucketAggregation.InternalBucket bucket : buckets) {
-            Map<String, Object> vars = new HashMap<>();
-            if (script.getParams() != null) {
-                vars.putAll(script.getParams());
+        if ("expression".equals(script.getLang())) {
```

Comment:
I don't think we should have any logic like this. Script uses should be agnostic to the script language implementation. Expressions need to handle returning a boolean value to satisfy this script's api.

Reply:
Expressions only know how to return doubles. I'm talking about the ExpressionScriptEngine, which is what implements each concrete class for a ScriptContext.

Reply:
@rjernst ah got it :) thanks => On it adjusting this :)

Reply:
done in 
https://github.com/elastic/elasticsearch/pull/32068/commits/0f13633790cd98fe64a507502bc79f1354c5d603, this is what we want right?

PS: Sorry for the split commits on this, the removal of the conditional by language is here: 
https://github.com/elastic/elasticsearch/pull/32068/commits/bc2746643a94f91c50af2680e5f4d49c65e20857